cmake/ctest/drivers/atdm/ats2/drivers/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt.sh

```diff
@@ -2,4 +2,16 @@
 if [ "${Trilinos_TRACK}" == "" ] ; then
   export Trilinos_TRACK=Experimental
 fi
+export TPETRA_ASSUME_CUDA_AWARE_MPI=0
 $WORKSPACE/Trilinos/cmake/ctest/drivers/atdm/ats2/local-driver.sh
+
+export TPETRA_ASSUME_CUDA_AWARE_MPI=1
+export CTEST_BUILD_NAME=Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_cuda-aware_static_dbg
+export CTEST_DO_UPDATES=FALSE
+#export CTEST_DO_NEW_START=FALSE
+export CTEST_START_WITH_EMPTY_BINARY_DIRECTORY=FALSE
+export CTEST_DO_CONFIGURE=FALSE
+export CTEST_DO_BUILD=FALSE
+atdm_run_script_on_compute_node \
```

Comment:
NOTE: The second call for CUDA-aware MPI makes no sense if you are not doing a CUDA build.  We need a way to handle that.  Therefore, we will need a way to handle that.  One way would be to source `local-driver.sh` instead of call it and with some other refactoring we have access to the `ATDM_CONFIG_` env vars generated from the parsed build name and then we can use this to set BUILD_NAME and decide if we want to run the second call.   Should not be too hard to do

Reply:
This driver is specifically for cuda builds, right?

Reply:
> This driver is specifically for cuda builds, right?

Got it.  Thanks for pointing that out.

I think we can refactor to move logic like this into Trilinos/cmake/ctest/drivers/atdm/ats2/local-driver.sh for all CUDA build to automatically run non-CUDA-aware and CUDA-aware MPI test suite.  But we can do that after we get this one build to work correctly.

Thanks!

Reply:
No problem! Agreed :)

Reply:
It's worth pointing out:

It would be nice if we wanted to setup testing on different platform, then all we needed to do is provide a suitable launch wrapper. (On Ride/White/Waterman, such a wrapper would would disable cuda-aware rather than enable)

IMO, that would be a good goal to keep in mind.

If the logic can avoid 'cuda' specific stuff, that would be a good long term strategy. (i.e., there will probably be other accelerators out there).  Of course, by then, the vendors may have eliminated the concept of 'device-aware' and 'regular' MPI (hopefully they will provide options to turn things off, and then we could still do these types of tests)